static/src/js/containers/DownloadHistoryContainer/DownloadHistoryContainer.jsx

```diff
@@ -1,71 +1,101 @@
-  retrievalHistory: state.retrievalHistory.history,
-  retrievalHistoryLoading: state.retrievalHistory.isLoading,
-  retrievalHistoryLoaded: state.retrievalHistory.isLoaded
-})
+export const DownloadHistoryContainer = ({
+  authToken,
+  earthdataEnvironment,
+  dispatchHandleError
+}) => {
+  const [retrievalHistory, setRetrievalHistory] = useState([])
+  const [retrievalHistoryLoading, setRetrievalHistoryLoading] = useState(false)
+  const [retrievalHistoryLoaded, setRetrievalHistoryLoaded] = useState(false)
 
-export const mapDispatchToProps = (dispatch) => ({
-  onFetchRetrievalHistory: (earthdataEnvironment) => {
-    dispatch(actions.fetchRetrievalHistory(earthdataEnvironment))
-  },
-  onDeleteRetrieval:
-    (retrievalId) => dispatch(actions.deleteRetrieval(retrievalId))
-})
+  const fetchRetrievalHistory = async () => {
+    if (!authToken) return
 
-export class DownloadHistoryContainer extends Component {
-  componentDidMount() {
-    const {
-      onFetchRetrievalHistory
-    } = this.props
+    setRetrievalHistoryLoading(true)
 
-    onFetchRetrievalHistory()
+    try {
+      const requestObject = new RetrievalRequest(authToken, earthdataEnvironment)
+      const response = await requestObject.all()
+      const { data } = response
+
+      setRetrievalHistory(data)
+      setRetrievalHistoryLoaded(true)
+    } catch (error) {
+      dispatchHandleError({
+        error,
+        action: 'fetchRetrievalHistory',
+        resource: 'retrieval history',
+        verb: 'fetching',
+        notificationType: 'banner'
+      })
+    } finally {
+      setRetrievalHistoryLoading(false)
+    }
   }
 
-  render() {
-    const {
-      earthdataEnvironment,
-      retrievalHistory,
-      onDeleteRetrieval,
-      retrievalHistoryLoading,
-      retrievalHistoryLoaded
-    } = this.props
-
-    return (
-      <DownloadHistory
-        earthdataEnvironment={earthdataEnvironment}
-        retrievalHistory={retrievalHistory}
-        retrievalHistoryLoading={retrievalHistoryLoading}
-        retrievalHistoryLoaded={retrievalHistoryLoaded}
-        onDeleteRetrieval={onDeleteRetrieval}
-      />
-    )
+  const handleDeleteRetrieval = async (retrievalId) => {
+    if (!authToken) return
+
+    try {
+      const requestObject = new RetrievalRequest(authToken, earthdataEnvironment)
+      await requestObject.remove(retrievalId)
+
+      setRetrievalHistory((prevHistory) => prevHistory.filter((item) => item.id !== retrievalId))
+
+      addToast('Retrieval removed', {
+        appearance: 'success',
+        autoDismiss: true
+      })
+    } catch (error) {
+      dispatchHandleError({
+        error,
+        action: 'handleDeleteRetrieval',
+        resource: 'retrieval',
+        verb: 'deleting',
+        notificationType: 'banner'
+      })
+    }
   }
-}
 
-DownloadHistoryContainer.defaultProps = {
-  retrievalHistory: []
+  useEffect(() => {
+    fetchRetrievalHistory()
+  }, [earthdataEnvironment, authToken])
```

Comment:
A few questions on this one:
- Are these necessary in the dependency array? 
- Should we prevent the function from firing if they are not defined?

`fetchRetrievalHistory` might need to be added here if its made a `useCallback`